generate/resources/couchbase-edge-server/README.md

```diff
@@ -1,16 +1,108 @@
+
+# QuickStart with Edge Server and Docker
+
+## Running Edge Server with Docker
+
+```
+$ docker run -d --name edge-server -p 59840:59840 couchbase/edge-server
+```
+
+At this point you should be able to send an HTTP request to the Edge Server on port 59840 using curl:
+
+```
+$ curl http://localhost:59840
+{"couchdb":"Welcome","vendor":{"name":"CouchbaseEdgeServer","version":"0.0.0 ()"},"version":"CouchbaseEdgeServer/0.0.0 () CouchbaseLiteCore/1.0.0 (21)"}
+```
+
+This setup is minimal and launches with an empty database, though any changes will be persisted.  Some key container paths are noted in the following diagram:
+
+![Default Docker Layout](./diagrams/docker-nomount.png)
+
+Under the `/opt/couchbase-edge-server` root, the bin and lib directories contain the necessary executable data to run the server, `etc/config.json` is the configuration file that was used at startup, and `var/databases` stores the resulting database specified by the configuration.
+
+## Viewing Logs
+You can view the Edge Server logs via the `docker logs` command:
+
+```
+$ docker logs edge-server
+2025-01-24T01:59:00.276Z        Using server TLS certificate: CN=127.0.0.1
+2025-01-24T01:59:00.277Z        Starting Couchbase Edge Server 0.0.0 ()
+2025-01-24T01:59:00.339Z        Sharing database https://localhost:59840/db/ from file /opt/couchbase-edge-server/var/databases/example.cblite2
+etc ...
+```
+
+# Customizing Edge Server configuration
+
+## Using a Docker volume
 
-# Running Edge Server with Docker
+The default configuration is useful for trying out the server, but a production deployment is going to be a little more involved in its creation.  A production deployment should contain persistent files that are outside of the container, as shown in the following diagram:
+
+![Production Docker](./diagrams/docker-mount.png)
+
+**Step - 1 :** Prepare a local working directory somewhere on your machine.  These steps will refer to your working directory as `$LOCALPATH`.  Inside this directory, create two folders called `etc` and `databases`.
+
+**Step - 2 :** Start the docker container in interactive mode with the previously created `etc` directory mounted.  This is accomplished by setting the entry point of the container to `bash`. 
 
-1. Start Edge Server container with the default configuration file
 ```
-$ docker run --name edge-server couchbase/edge-server
+docker run -it --rm --entrypoint=bash -v $LOCALPATH/etc:/opt/couchbase-edge-server/etc couchbase/edge-server
 ```
 
-2. Start Edge Server container with an external configuration file
+**Step - 3 :** Create a TLS key and certificate for use with the server.  Normally, this would be done through your provider and the instructions are out of scope for this README, but for testing and tutorial purposes the edge server can create a self signed certificate for you.  The following command will create a cert and key file in the mounted `etc` directory (the default working directory of the container):
```

Comment:
`Normally, this would be done through your provider and the instructions are out of scope for this README`
where can an user find such information?  docs.couchbase.com (even thought it doesn't exist yet)?  If so, it would be good to make a reference in the readme.

Reply:
No, they would find it through whoever their certificate provider is.  Let's Encrypt, cloudflare, etc.